packages/server/src/index.ts

```diff
@@ -2,62 +2,150 @@ import { McpServer } from '@modelcontextprotocol/sdk/server/mcp.js';
 import { StdioServerTransport } from '@modelcontextprotocol/sdk/server/stdio.js';
 import { CredoToolKit } from '@cheqd/mcp-toolkit-credo';
 import * as dotenv from 'dotenv';
+import { ToolDefinition } from '@cheqd/mcp-toolkit-credo/build/types.js';
 
 dotenv.config();
 
-const port = process.env.PORT || 3000;
-
-class AgentMcpServer {
-	server: McpServer;
+/**
+ * AgentMcpServer extends McpServer to provide specialized functionality
+ * for the cheqd agent, including tool setup, signal handling, and proper cleanup.
+ */
+class AgentMcpServer extends McpServer {
+	private transport: StdioServerTransport | null = null;
+	private credoToolkit: CredoToolKit | null = null;
 
+	/**
+	 * Initialize the server with a name and version, and set up capabilities
+	 */
 	constructor() {
-		this.server = new McpServer({
-			name: 'cheqd-mcp-toolkit-server',
-			version: '1.0.0',
-		});
+		super(
+			{
+				name: 'cheqd-mcp-toolkit-server',
+				version: '1.0.0',
```

Comment:
Isn't there a way to insert/use the version in package.json or automatically picked up from Docker image tag?